editor/src/components/navigator/navigator-item/navigator-item.tsx

```diff
@@ -329,7 +333,13 @@ export const NavigatorItem: React.FunctionComponent<NavigatorItemInnerProps> = b
     })
 
     return (
-      <FlexRow ref={domElementRef} style={rowStyle} onMouseDown={select} onMouseMove={highlight}>
+      <FlexRow
+        ref={domElementRef}
+        style={rowStyle}
+        onMouseDown={select}
+        onMouseMove={highlight}
+        onDoubleClick={focusComponent}
+      >
```

Comment:
I think this is fine but it leaves a problem where double clicking different parts of the row have very different actions. IMO if we're going to support double clicking to focus (which definitely feels right to me), it should apply to the label as well, and we should change renaming to be an explicit action triggered by the context menu

Reply:
I recommend you to try it first, to me it was surprisingly uncontentious that double clicking can lead to two actions – but of course I knew where to click

Reply:
I think that's the thing, you know where to click. I've seen this pattern in a few different apps and I've always found it irritating tbh.